### PR TITLE
[Refactor, Breaking] Changes in Observable and Scale

### PR DESCRIPTION
**Pull request by chMoussa**
Related to issue #192.

To complete after https://github.com/pasqal-io/pyqtorch/pull/220

- [x] Rename run and forward for forward and expectation in observables.
- [x] More streamlined Scale class.
- [ ] More tests.

### CODE REVIEW

**Review by jpmoutinho — DISMISSED** [review]
Thanks @chMoussa 👍

This is a small thing but now is the time to iron out these conceptual details.

Let's wait for @dominikandreasseitz approval btw. Unfortunately this is a breaking change, so we should think carefully about the issues upstream.

File: pyqtorch/analog.py
Comment (dominikandreasseitz) [inline pyqtorch/analog.py:365]: its a bit strange to me that the forward of an  Add block does not add but just do the normal sequence forward

Comment (jpmoutinho) [reply pyqtorch/analog.py:365]: Sorry, you are right. The point was actually that the `forward` of the observable should be the same as the `forward` from `Add`, so actually there should no be `forward` rewrite on the `Observable` because now it inherits from `Add`.

The `DiagonalObservable` currently inherits from `Primitive` so it should be handled differently.

Comment (chMoussa) [reply pyqtorch/analog.py:365]: I see what you mean. Then it is better to remove the forward method to make use the one from Add.

Comment (chMoussa) [reply pyqtorch/analog.py:347]: Now this should be handled @dominikandreasseitz @jpmoutinho.

File: pyqtorch/analog.py
Comment (jpmoutinho) [inline pyqtorch/analog.py:344]: This is ok for now to replicate the previous logic but this class will need to be reworked, essentially so we can deal with https://github.com/pasqal-io/pyqtorch/issues/175

I also don't think we need a special `DiagonalObservable` class, just a property that keeps track of the diagonalily of the individual blocks when they are composed of `Z`, `I` or `N` and a version of `apply_operator` specific for diagonal operations. This can be used throughout PyQ for applying diagonal evolutions, and also for applying diagonal gates like `CPHASE`.

But I would not do this in this MR.

**Review by jpmoutinho — DISMISSED** [review]
Thanks @chMoussa! Now I think this is ok :)

File: pyqtorch/analog.py
Comment (chMoussa) [reply pyqtorch/analog.py:280]: Sure, created issue #211 for handling this.

**Comment by jpmoutinho** [comment]
Closing as the changes from this MR are now done in https://github.com/pasqal-io/pyqtorch/pull/220

**Pull request closed without merging**
